### ISSUE TITLE
[Regression] Version above 2.0.4370 report Error: Collision detected inserting into object: x-ms-parameter-location

### ISSUE THREAD

**Issue by XiangyuL-Microsoft** [Azure/autorest#3245]
By run the autorest cli:

> autorest --input-file=mixedreality.json --azure-arm --csharp --clear-output-folder --version==2.0.4370

against a [swagger file](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/mixedreality/resource-manager/Microsoft.MixedReality/preview/2019-02-28-preview/mixedreality.json) which exists for a long time. The tool report certain error:

```
FATAL: openapi-document/openapi-document-converter - FAILED
FATAL: Error: Collision detected inserting into object: x-ms-parameter-location
Process() cancelled due to exception : Collision detected inserting into object: x-ms-parameter-location
  Error: Collision detected inserting into object: x-ms-parameter-location
```

Version 2.0.4371 also has this problem, but versions below and include 2.0.4302 are OK.

**Comment by srmorin** [comment]
We are seeing the same issue.  2.0.4302 is fine, 2.0.4371 fails.

**Comment by GaryMalk** [comment]
I get the error with 2.0.4302, as well.  Verbose mode:
FATAL: openapi-document/openapi-document-converter - FAILED
FATAL: Error: Collision detected inserting into object: constructor
Process() cancelled due to exception : Collision detected inserting into object: constructor
  Error: Collision detected inserting into object: constructor

**Comment by NelsonDaniel** [comment]
@GaryMalk, @srmorin, @XiangyuL-Microsoft  This issue has been resolved from 2.0.372+
